src/main/java/com/stuypulse/robot/subsystems/Climber.java

```diff
@@ -5,24 +5,155 @@
  * Contains:
- *      - TBD
+ *      - Change tilt of climber
+ *      - Move climber via motor
+ *      - Different tilt angles
+ *      - Encoder + Solenoid used for stopping
  *
- * @author [TBD]
+ * @author independence106(Jason Zhou)
+ * @author Ca7Ac1(Ayan Chowdhury)
+ * @author marcjiang7(Marc Jiang)
+ * @author ambers7(Amber Shen)
+ * @author Souloutz(Howard Kong)
+ * @author jiayuyan0501(Jiayu Yan)
+ * @author ijiang05(Ian Jiang)
+ * @author TraceyLin (Tracey Lin)
+ * @author annazheng14(Anna Zheng)
+ * @author lonelydot(Raymond Zhang)
+ * @author andylin2004 (Andy Lin)
  */
 public class Climber extends SubsystemBase {
 
+    private Solenoid solenoidLong;
+    private Solenoid solenoidShort;
+
+    private DigitalInput limitSwitch;
+
+    private Solenoid stopper;
+
+    private RelativeEncoder encoder;
+
+    private CANSparkMax climber;
+
     public Climber() {
-        // Initialize Motors, etc... here!
+        climber = new CANSparkMax(Constants.Ports.Climber.MOTOR);
+        climber.setInverted(Constants.ClimberSettings.MOTOR_INVERTED);
+  
+        encoder = climber.getEncoder();
+        encoder.setPositionConversionFactor(Constants.ClimberSettings.CIRCUMFERENCE * Constants.ClimberSettings.GEAR_RATIO);
+
+        solenoidLong = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.SOLENOID_LONG);
+        solenoidShort = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.SOLENOID_SHORT);
+        stopper = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.STOPPER);
+
+        limitSwitch = new DigitalInput(Constants.Ports.Climber.LIMIT_SWITCH);
+    }
+    
+    private void moveMotor(double speed) {
+        climber.set(speed);
+    }
+
+    public void liftUp() {
+        moveMotor(Constants.ClimberSettings.CLIMBER_DEFAULT_SPEED.get());
+    }
+
+    public void liftDown() {
+        moveMotor(-Constants.ClimberSettings.CLIMBER_DEFAULT_SPEED.get());
+    }
+
+    public void stop() {
+        moveMotor(0.0);
+    }
+    
+    public void liftUpSlow() {
+        moveMotor(Constants.ClimberSettings.CLIMBER_SLOW_SPEED.get());
+    }
+
+    public void liftDownSlow() {
+        moveMotor(-Constants.ClimberSettings.CLIMBER_SLOW_SPEED.get());
+    }
+
+    public void extendLong() {
+        solenoidLong.set(true);
+    }
+    
+    public void retractLong() {
+        solenoidLong.set(false);
+    }
+    
+    public void extendShort() {
+        solenoidShort.set(true);
+    }
+
+    private void retractShort() {
+        solenoidShort.set(false);
+    }
+
+    public void fullyRetract() {
+        retractShort();
+        retractLong();
+    }
+
+    public void fullyExtend() {
+        extendLong();
+        extendShort();
+    }
+
+    public boolean getLongExtended() {
+        return solenoidLong.get();
+    }
+
+    public boolean getShortExtended() {
+        return solenoidShort.get();
+    }
+
+    public double getEncoderDistance() {
+        return encoder.getPosition();
```

Comment:
what is the significance of this encoder? it should be reflected in the name. (e.g. climber height? depth?) 

also if it is for climber height consider having another method that divides this value by the max climber height -- i imagne that working with 0,1 could be a lot easier